packages/bvaughn-architecture-demo/components/console/MessagesList.tsx

```diff
@@ -98,14 +98,28 @@ function MessagesList({ forwardedRef }: { forwardedRef: ForwardedRef<HTMLElement
           />
         );
       } else if (isPointInstance(loggable)) {
-        listItems.push(
-          <LogPointRenderer
-            key={index}
-            index={index}
-            isFocused={loggable === currentSearchResult}
-            logPointInstance={loggable}
-          />
-        );
+        // Respect the focus region and _don't_ show LogPointRenderers  outside the focus range
+        // (since they have analysis that results in errors otherwise ).
+        // TODO Is this _too_ restrictive?
+
+        const isFocused =
+          focusRange === null ||
+          isExecutionPointsWithinRange(
+            loggableExecutionPoint,
+            focusRange.begin.point,
+            focusRange.end.point
+          );
```

Comment:
Why would we only need to do this for log points?

If there is an issue where we are showing things outside of the focused region, wouldn't that apply to _all types of messages_ (in the same way as the `isLoaded` check above does) ?

Reply:
> Why would we only need to do this for log points?

Answer: It looks like other log types are being pre-filtered by the `LoggablesContextRoot`, e.g.
```js
// Trim eventLogs and logPoints by focusRange.
// Messages will have already been filtered from the backend.
const focusedEventLogs = useMemo<EventLog[]>(() => {
  if (focusRange === null) {
    return eventLogs;
  } else {
    return eventLogs.filter(eventLog =>
      isExecutionPointsWithinRange(eventLog.point, focusRange.begin.point, focusRange.end.point)
    );
  }
}, [eventLogs, focusRange]);
```

This is also expected to occur for the log points:
```
      const [hitPoints, status] = getHitPointsForLocation(
        client,
        point.location,
        point.condition,
        focusRange
      );
```

But for some reason that filtering doesn't seem to be working correctly for us in all scenarios? I think we should dig into that.

Reply:
The code that's running here is:
```js
const [executionPoints, status] = await client.getHitPointsForLocation(
  focusRange,
  location,
  condition
);
```

So why is it giving us back analysis points that are outside of the focus region?

Reply:
This is the problem:
https://github.com/replayio/devtools/blob/84892c90085993a053f80ba399ca7e8705fb01b9/packages/bvaughn-architecture-demo/src/suspense/PointsCache.ts#L48-L54

This "key" string will just be "...:[Object]:[Object]" for any focus range.